static/js/profile.js

```diff
@@ -0,0 +1,645 @@
+					$('#pe_status').hide().text('Error. Default values used.').fadeIn("slow");
+				}
+			}).done(initeditor);
+		});
+	} else {
+		$.ajax('/api/v1/profile/current', {
+				success: function (record) {
+					c_profile = record;
+					}
+		});
+	}
+	
+	function initeditor() {
+		// Add handler for style switching
+		$('#pe_input_simple').change(switchStyle);
+		$('#pe_input_calculator').change(switchStyle);
+
+		// display status
+		$('#pe_units').text(apistatus['defaults']['units']);
+		$('#pe_timeformat').text(apistatus['defaults']['timeFormat']+'h');
+		$('#pe_title').text(apistatus['defaults']['customTitle']);
+
+		var lastvalidfrom = new Date(mongoprofiles[1] && mongoprofiles[1].validfrom ? mongoprofiles[1].validfrom : null);
+		
+		//timepicker
+		$('#pe_validfrom').datetimepicker({
+			  lang: apistatus['defaults']['language']
+			, mask: true
+			, stepMinute: 15
+			, timeFormat: apistatus['defaults']['timeFormat'] == '12' ? 'hh:mm p' : 'HH:mm'
+			, dateFormat: apistatus['defaults']['timeFormat'] == '12' ? 'mm/dd/yy' : 'd.m.yy'
+			, minDate: lastvalidfrom
+			//this has a serious bug
+			//, minDateTime: mongoprofiles[1] && mongoprofiles[1].validfrom ? mongoprofiles[1].validfrom : null
+			, maxDate: new Date()
+			, maxDateTime: new Date()
+			, onSelect: function(d,i){
+				  if(d !== i.lastVal){
+					  $(this).change();
+				  }
+			 }
+		}).on('change', dateChanged);
+
+		//$('#pe_validfrom').datetimepicker('setDate', lastvalidfrom);
+	
+		// Set values from profile to html
+		fillFrom();
+		// hide unused style of ratios
+		switchStyle();
+		// show proper submit button
+		dateChanged();
+		
+		// date of last record
+		if (lastvalidfrom)
+			$('#pe_lastrecvalidfrom').html('Last record date: '+lastvalidfrom.toLocaleString()+' <i>(Date must be newer to create new record or the same to update current record)</i>');
+		else 
+			$('#pe_lastrecvalidfrom').html('');
+		
+		console.log('Done initeditor()');
+	}
+	
+	// Handling valid from date change
+	function dateChanged(event) {
+		var newdate = $('#pe_validfrom').datetimepicker('getDate');
+		if (mongoprofiles.length<2 || +new Date(mongoprofiles[1].validfrom) != +newdate) {
+			$('#pe_submit').text('Create new record');
+			$('#pe_validfrom').css({'background-color':'green'});
+			$('#pe_submit').css({'background-color':'green'});
+		} else {
+			$('#pe_submit').text('Update record');
+			$('#pe_validfrom').css({'background-color':'white'});
+			$('#pe_submit').css({'background-color':'buttonface'});
+		}
+		if (event) {
+			event.preventDefault();
+		}
+	}
+	
+	// Handling html events and setting/getting values
+	function switchStyle(event) {
+		if ($('#pe_input_simple').prop('checked')) {
+			$('#pe_simple').show("slow");
+			$('#pe_calculator').hide("slow");
+		} else {
+			$('#pe_simple').hide("slow");
+			$('#pe_calculator').show("slow");
+		}
+		if (event) {
+			event.preventDefault();
+		}
+	}
+	
+	function fillFrom(event) {
+		var mmoltime = [
```

Comment:
why do we need fixed times, pump UI's are like this, but we have way more flexibility